build.sbt

```diff
@@ -134,101 +66,70 @@ def scalaVersionSpecificFolders(srcName: String, srcBaseDir: java.io.File, scala
     List(CrossType.Pure, CrossType.Full)
       .flatMap(_.sharedSrcDir(srcBaseDir, srcName).toList.map(f => file(f.getPath + suffix)))
   CrossVersion.partialVersion(scalaVersion) match {
-    case Some((2, y))     => extraDirs("-2.x") ++ (if (y >= 13) extraDirs("-2.13+") else Nil)
-    case Some((0 | 3, _)) => extraDirs("-2.13+") ++ extraDirs("-3.x")
+    case Some((2, y))     => if (y >= 13) extraDirs("-2.13+") else Nil
+    case Some((0 | 3, _)) => extraDirs("-2.13+")
     case _                => Nil
   }
 }
 
-ThisBuild / mimaFailOnNoPrevious := false
-
-def doctestGenTestsDottyCompat(isDotty: Boolean, genTests: Seq[File]): Seq[File] =
-  if (isDotty) Nil else genTests
-
 lazy val commonSettings = Seq(
-  scalacOptions ++= commonScalacOptions(scalaVersion.value, isScala3.value),
   Compile / unmanagedSourceDirectories ++= scalaVersionSpecificFolders("main", baseDirectory.value, scalaVersion.value),
   Test / unmanagedSourceDirectories ++= scalaVersionSpecificFolders("test", baseDirectory.value, scalaVersion.value),
-  resolvers ++= Seq(Resolver.sonatypeRepo("releases"), Resolver.sonatypeRepo("snapshots")),
-  Test / parallelExecution := false,
-  testFrameworks += new TestFramework("munit.Framework"),
+  Compile / packageSrc / mappings ++= {
+    val base = (Compile / sourceManaged).value
+    (Compile / managedSources).value.map { file =>
+      file -> file.relativeTo(base).get.getPath
+    }
+  },
+  scalacOptions ~= { _.filterNot(x => x.startsWith("-Wunused:")) },
```

Comment:
This was the most disappointing thing. Seems that Cats has got lots and lots of unused warnings. So we have to disable them if we want fatal warnings for everything else. It would truly be a great thing if someone could chase down and at least `@nowarn` them if they are legit or necessary for bincompat. Then at least we can avoid adding more unused stuff.

Reply:
Seems that Scala 3.0.2 ignores any `@nowarn` annotations, unfortunately. I found it out while working on #4083.

Reply:
Yes, that's right. But`@nowarn` is supported in Scala 3.1.0 and I believe we should bump for that and other reasons. See:
- https://github.com/typelevel/scalacheck/issues/878

Reply:
We won't be getting any "unused" warnings anymore after this change, is that correct?

Reply:
That's right, no more unused warnings :(

Reply:
Hmm... Wouldn't it better to disable the `-Werror` option instead (along with `-Xfatal-?whatever?` used in Scala 2.12)? In this case we will still be seeing warning and hopefully one day will have them fixed :)
Wdyt?

Reply:
It's a good question. So IIUC, you mean enable all warnings, but disable fatal warnings?

The only problem is that there are so many unused warnings they drown out everything else maybe.

Reply:
Yes, I also hate seeing warning:) This is why I attempted to make some relief in #4083. And I would be glad to continue working on that.

Answering your question – yes, I think seeing warnings is still better than suppressing them completely.

Reply:
Yes, now that I came face-to-face with the problem myself, I truly appreciate your heroic efforts in https://github.com/typelevel/cats/pull/4083 :)

Ok, that's fair. I'll restore all the default compiler flags from sbt-typelevel, but disable fatal warnings.

Reply:
Btw: there're also loads of deprecation warnings in some modules (which mostly don't make any real sense). And there're so many of them that it is quite easy to miss something really important.

Reply:
Right, that's exactly my concern. The only way to be sure you don't miss warnings, is to make them fatal. But that means we have to disable the warnings that we can't solve today.